scarf-sh/hackage-server
# alex/fix analytics pixels

- Drive-by shooting down an unused import
- Commonmark: gfmExtensions needs to precede the defaultSyntaxSpec (#1000,#1001)
- Fix compilation with haddocks
- Re #1005: Restore haddock module header of RequestContentTypes
- Up CI to cabal 3.6, add haddock step
- README: cosmetics, formatting
- tests/Util: remove `die`
- Cosmetics: remove extra parentheses
- Fix whitespace: trailing spaces, trailing empty lines, tabs
- Configuration file for fix-whitespace
- Dependency time-locale-compat can be eliminated since time >= 1.6
- Remove #if-conditionals that predate GHC 7.10 and time-1.5
- Removed unused import
- Cosmetical fixes that help with doctest
- Suppress unused-record-wildcards warning (new in GHC 8.10)
- Include cabal.* in fix-whitespace check
- Rearrange some definitions in preparation for GHC 9.0 Template Haskell
- Fix parsing of ISO times
- Update README for cabal configuration clarity
- Unit tests for renderMarkdown in doctest format (#1000, #1001)
- Force .ghc.environment files so that doctests work with `cabal test`
- Move library code into subdirectory src/
- Add new-browse on different endpoint
- Make new-browse handle old endpoints
- Adjust PaginationTest for pageSize=50
- Use flexbox instead of float
- Use max-width instead of min-width for #left-pane
- Scroll module list instead of breaking words
- Long module names push down properties box
- Width of #content varies, so #left-pane children also vary
- Add JSON endpoints for basic package information
- Prettier/consistent margins in and around the form
- update DataTables CSS and JS
- Doctest running via `doctest-parallel`
- Simplify DocTestMain thanks to doctest-parallel-0.2.2
- Vendor ListT, removed in transformers-0.6 (Fix #1010) (#1027)
- GHC 9.0 migration: code rearragements (#1016)
- Cabal 3.4: bump to Cabal-3.4, removed VersionRangeF cases (#1016)
- Cabal 3.4: copy D.S.Util.CabalRevisions from hackage-cli (#1016)
- Cabal 3.4: mainLibSet (new smart constructor) (#1016)
- Cabal 3.4: Flag -> PackageFlag (#1016)
- Cabal 3.4: need to import D.Types.ModuleReexport (#1016)
- Cabal 3.4: KnownRepoType (reorganized data type) (#1016)
- Cabal 3.4: FieldGrammar (#1016)
- Cabal 3.4: CabalSpecVersion (#1016)
- CI: include GHC 9.0 (#1016)
- typo
- typos
- Enable filtering packages by distro
- cleanup based on comments about outstanding work
- Rename 'new-browse' to 'browse', remove dead search explanation code
- Cosmetics: unobfuscate accumJPath
- Migrate to aeson-2.0
- Minimal migration to Cabal-3.6.3.0
- Build with GHC 9.2
- PackageTestMain: comments, cosmetical changes
- PackageTestMain: factor out successTest
- Fix #1030: test case
- aeson fix
- force no cache delay for tests
- Link build report from package page
- Simplify getting all package names and use `itemCache` in Browse (#1049)
- Restore two-column layout for very wide viewports (#1050)
- Admins can view `/maintain` page (#1045)
- Small BuildClient fixes (#1044)
- Helpful error message on missing group membership (#1054)
- Default to HTML handler, not the first (#1053)
- Fix some hlint warnings (#1056)
- Fix some hlint warnings - part 2 (#1057)
- Add instructions for install native deps for Fedora
- Auto-generate more complete CI via haskell-ci (#1061)
- Fix incomplete-record-updates warnings (#1059)
- Use assert{Bool,Failure} from tasty-hunit
- Add search/browse form for users without JavaScript (#1052)
- Avoid top-level await, QuteBrowser compat (#1062)
- Add form to update full name and email of user (#1047)
- Introduce analytics pixels infrastructure (#1042)
- cleanup text from last pr
- comment out pixels until fixed
- update `nix-shell` sources; GitHub Actions for `nix-shell` (#1002)
- fix(server): email address validation (#1074)
- delete unused GitHub Action; fix CI badge
- Fix #1076: separate validators from UI and doctest them (#1077)
- Cachix caching for nix-shell GitHub Action (#1081)
- Add uploaded_at field in package api (#1080)
- package page: Include virtual-modules in module tree (#1085)
- Fix analyticsPixel template variable name
